docs/developers-guide/decorators/json-schema-enabled decorator.md

```diff
@@ -0,0 +1,93 @@
+# The json-schema-enabled Decorator
+This document explains how the json-schema-enabled decorator works and how it is applied to:
+* values
+* types
+* modules
+It also explains how a Json Schema is generated based on the annotation
+
+## Overview of json-schema-enabled Decorator
+The json-schema-enabled decorator is modeled as a boolean type in the JsonSchema module in the
+Attribute model.
+When a json-schema-enabled decorator value of true is applied to a type or module, then json-schema
+backend code generation would be based on these. 
+When applied to a type or value, the fully qualified name is used as the key in the attributed dictionary.
+But when applied to a module, the module name is used as the key
+All decorator configurations for a project are place in the attributes-conf.json file placed next to the main IR.
+
+## Custom Configuration Modeling
+The Custom Attribute configuration model is found in the [CustomAttributes](../../../src/Morphir/CustomAttribute) directory.
+The main types of for the decorator configuration model are the CustomAttributeId,
+the CustomAttributeConfig, the CustomAttributesConfigs, the CustomAttributesInfo
+and the CustomAttributesDetails
+* CustomAttributeId - a string type that represents the unique id of the decorator
+* CustomAttributeConfig - a record type with one field: filePath which is of type FilePath and represents the attributes-conf.json file
+* CustomAttributesConfigs - dictionary of all CustomAttributeConfigs keyed by the CustomAttributeID
+* CustomAttributeInfo - dictionary of CustomAttributeInfo keyed by the CustomAttributesId
+* CustomAttributeDetails - a record type of with the following fields:
+  * displayName - String display name of  the custom attribute
+  * entryPoint - FQName or ModuleName being decorated
+  * iR - Distribution for the custom attribute IR
+  * data - the custom attribute dictionary (content of the <unique-id.json> file)
```

Comment:
This whole section looks more like a contributor doc then something for the user. The user does not think in terms of types. They want  to create a JSON file that works.

Reply:
I have renamed this guide to json-schema-enabled developers guide. There's another less detailed document created as the user's guide.